suzieq/engines/pandas/interfaces.py

```diff
@@ -89,6 +87,69 @@ def get(self, **kwargs):
         else:
             return df.reset_index(drop=True)[fields]
 
+    def _check_vlan_match(self, vlan_list: List[str],
+                          df: pd.DataFrame) -> pd.DataFrame:
+        '''Return a dataframe with rows in VLANs requested
+
+        VLAN is treated specially because its the only field that is a list
+        of integers, and can be filtered on multiple fields: vlan and vlanList
+        A user filter on VLAN is a list that can contain numeric comparisons
+        as well as simple integers.
+
+        Its not made into a generic filter match because this match on VLAN
+        applies only to interface table.
+        '''
+
+        # If there are any VLANs with ! mixed without !, remove the
+        # ! ones. vlanList is an OR list, not an AND list, except
+        # if all are !
+        if any(x.startswith('!') for x in vlan_list):
+            cond = 'and'
+        else:
+            cond = 'or'
+
+        resdf_list = []
+        if (any(x.startswith('>') for x in vlan_list) and
```

Comment:
Wondering whether this is the right thing to do. 
It is true in the case of a condition like `vlan > 10 vlan < 20` we need an `and`, but it also true that if I want something like `vlan < 10 vlan > 20` I need an `or`. 
Can't we find a way to make things more explicit? 
Maybe we can ask to specify `and` or `or` or we might define an operator for intervals, 
for example id we want an interval we can say `[10, 20]` for `10 <= vlan <= 20`  or `]10, 20]` for `10 < vlan <= 20`.